temba/channels/models.py

```diff
@@ -1105,6 +1129,48 @@ def success(cls, channel, msg, msg_status, start, request_method=None, request_u
                                   request_time=request_time)
 
     @classmethod
+    def send_fcm_message(cls, channel, msg, text):
+        from temba.msgs.models import Msg, WIRED
+        start = time.time()
+
+        api_key = channel.config.get(Channel.CONFIG_FCM_KEY) if Channel.CONFIG_FCM_KEY in channel.config else None
+
+        if api_key:
+            url = 'https://fcm.googleapis.com/fcm/send'
+            title = channel.config.get(Channel.CONFIG_FCM_TITLE)
+            data_message = {'type': 'rapidpro', 'title': title, 'message': text, 'message_id': msg.id}
+            payload = json.dumps({
+                'data': data_message,
+                'content_available': True,
+                'to': msg.urn_path,
+                'notification': {
+                    "title": title,
+                    'body': data_message['message']
+                },
+                'priority': 'high'
+            })
+            try:
+                headers = {'Content-Type': 'application/json', 'Authorization': 'key=%s' % api_key}
+                headers.update(TEMBA_HEADERS)
+                response = requests.post(url, data=payload, headers=headers, timeout=5)
+                result = json.loads(response.content)
+                if response.status_code == 200 and 'success' in result and result['success'] == 1:
+                    external_id = result.get('multicast_id')
+                    Channel.success(channel, msg, WIRED, start, 'POST', url, payload, response, external_id)
+                else:
+                    raise SendException("Got non-200 response [%d] from Firebase Cloud Messaging" %
+                                        response.status_code,
+                                        method='POST',
+                                        url=url,
+                                        request=payload,
+                                        response=result,
+                                        response_status=response.status_code)
+            except Exception as e:  # pragma: no cover
+                ChannelLog.log_error(msg, e)
```

Comment:
Good to log what was actually posted (and what any possible response was) here, aids in debugging.